src/advanced_inv.cpp

```diff
@@ -2038,7 +2038,8 @@ int advanced_inventory::add_item( aim_location destarea, item &new_item, int cou
         // @todo: Make it use same exact methods as regular pickup
         if( ( destarea == AIM_INVENTORY || destarea == AIM_WORN || panes[dest].in_vehicle() ) &&
             new_item.is_bucket_nonempty() &&
-            !query_yn( _( "Spill contents of %s?" ), new_item.tname().c_str() ) ) {
+            !query_yn( _( "Spill contents of %s?" ),
+                       colorize( new_item.tname().c_str(), new_item.color_in_inventory() ) ) ) {
```

Comment:
```suggestion
                       colorize( new_item.tname(), new_item.color_in_inventory() ) ) ) {
```
No need for the `.c_str()` (and similarly for many of your other changes).

Reply:
Right. Cleaned it.